keps/sig-auth/3766-referencegrant/README.md

```diff
@@ -441,174 +368,365 @@ rejected".
+different purposes.
 
-* Clarify that an implementation is required to reconcile ReferenceGrant for
-  specific `To` Kinds.
-* Corollary for future work, define how controllers interact. Is it a problem if
-  multiple controllers consume the same ReferenceGrant?
-* Status design is still pending, but it's currently expected that controllers
-  will indicate status on the _referring_ resources, not on ReferenceGrant itself.
-<<[UNRESOLVED do we need status? ]>>
-Original Thread: https://github.com/kubernetes/enhancements/pull/3767#discussion_r1084670421
+We will initially define 3 well-known values here to represent the current use
+cases of ReferenceGrant, and we will leave room for other custom domain-prefixed
+values if this API is used for implementation-specific use cases in the future.
 
-We want to be able to represent the following, in descending order of 
-importance: 
+Future standard verbs will be approved by SIG Auth. In general, standard verbs
+should be reserved for:
 
-1. Communicate that the ReferenceGrant is actively being used
-2. Communicate which controllers are using this ReferenceGrant
-3. Communicate how many times it's been used with sufficient granularity that I
-   can see the effects of my changes (if I remove this reference, am I removing
-   a dependency on this ReferenceGrant?)
+* References to and from official Kubernetes APIs
+* Well defined use cases that will have conformance tests
+* Concepts that will be broadly available in Kubernetes clusters
 
-We could introduce a status structure that allowed each implementing controller
-to write 1 entry:
+#### 4. Namespace Label Selectors
+Namespace Label Selectors add significant complexity for both users and implementors of
+this API. With that said, we believe they provide sufficient value to support as part
+of this API. We will add a new `from.namespace` struct that allows specifying either a
+name or a selector.
+
+**Before Change:**
+```yaml
+apiVersion: authorization.k8s.io/v1alpha1
+kind: ReferenceGrant
+metadata:
+  name: allow-baz-httproutes
+  namespace: quux
+from:
+- namespace: baz
+  group: gateway.networking.k8s.io
+  resource: httproutes
+to:
+- group: ""
+  resource: services
+```
 
+**After:**
 ```yaml
-status:
-  referencesAllowed:
-  - controllerName: gateway.example.com
-    numReferences: 1
+apiVersion: authorization.k8s.io/v1alpha1
+kind: ReferenceGrant
+metadata:
+  name: allow-baz-httproutes
+  namespace: quux
+from:
+  namespace:
+    name: baz
+  resources:
+  - group: gateway.networking.k8s.io
+    resource: httproutes
+to:
+- group: ""
+  resource: services
 ```
 
-@thockin responded with:
+This change was inspired by earlier feedback on the KEP:
+
+* https://github.com/kubernetes/enhancements/pull/3767#discussion_r1084492070
+* https://github.com/kubernetes/enhancements/pull/3767#discussion_r1084674648
+
+### Additional Considerations
+
+#### Using ReferenceGrant to Grant RBAC to Controllers
```

Comment:
Does this only cover RBAC? (if my cluster uses a different, custom authz mechanism, could the mechanism described here still apply)

```suggestion
#### Using ReferenceGrant to Grant Access to Controllers
```